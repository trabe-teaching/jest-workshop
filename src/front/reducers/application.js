import { combineReducers } from "redux";
import loading, * as fromLoading from "reducers/loading";
import error, * as fromError from "reducers/error";
import users, * as fromUsers from "reducers/users";

export default combineReducers({ loading, error, users });

export const isLoading = state => fromLoading.isLoading(state.loading);
export const hasError = state => fromError.hasError(state.error);
export const getLightSiders = state => fromUsers.getLightSiders(state.users);
export const getDarkSiders = state => fromUsers.getDarkSiders(state.users);
