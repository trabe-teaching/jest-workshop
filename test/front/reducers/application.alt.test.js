import deepFreeze from "deepfreeze";
import application, { isLoading, hasError, getLightSiders, getDarkSiders } from "reducers/application";

const state = {
  error: false,
  loading: 1,
  users: {
    repo: {
      1: { name: "john", actions: [] },
      2: { name: "lucas", actions: [] },
    },
    lightSiders: [1],
    darkSiders: [2],
  },
};

deepFreeze(state);

describe("front/reducers/application", () => {
  describe("reducer", () => {
    it("has a default state", () => {
      expect(application(undefined, { type: "@init" })).toMatchSnapshot();
    });

    it("reduces nothing", () => {
      expect(application(state, { type: "NOPE" })).toMatchSnapshot();
    });

    it("reduces REQUEST", () => {
      expect(application(state, { type: "XXXX_REQUEST" })).toMatchSnapshot();
    });
  });

  deepFreeze(state);

  describe("#isLoading", () => {
    expect(isLoading(state)).toBe(true);
  });

  describe("#hasError", () => {
    expect(hasError(state)).toBe(false);
  });

  describe("#getLightSiders", () => {
    expect(getLightSiders(state)).toMatchSnapshot();
  });

  describe("#getDarkSiders", () => {
    expect(getDarkSiders(state)).toMatchSnapshot();
  });
});
