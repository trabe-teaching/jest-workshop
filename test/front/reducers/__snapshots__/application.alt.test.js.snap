// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  Object {
    "canBanish": false,
    "canBetray": false,
    "canSeek": false,
    "canTurn": false,
    "master": undefined,
    "name": "john",
  },
]
`;

exports[` 2`] = `
Array [
  Object {
    "canBanish": false,
    "canBetray": false,
    "canSeek": false,
    "canTurn": false,
    "master": undefined,
    "name": "lucas",
  },
]
`;

exports[`front/reducers/application reducer has a default state 1`] = `
Object {
  "error": false,
  "loading": 0,
  "users": Object {
    "darkSiders": Array [],
    "lightSiders": Array [],
    "repo": Object {},
  },
}
`;

exports[`front/reducers/application reducer reduces REQUEST 1`] = `
Object {
  "error": false,
  "loading": 2,
  "users": Object {
    "darkSiders": Array [
      2,
    ],
    "lightSiders": Array [
      1,
    ],
    "repo": Object {
      "1": Object {
        "actions": Array [],
        "name": "john",
      },
      "2": Object {
        "actions": Array [],
        "name": "lucas",
      },
    },
  },
}
`;

exports[`front/reducers/application reducer reduces nothing 1`] = `
Object {
  "error": false,
  "loading": 1,
  "users": Object {
    "darkSiders": Array [
      2,
    ],
    "lightSiders": Array [
      1,
    ],
    "repo": Object {
      "1": Object {
        "actions": Array [],
        "name": "john",
      },
      "2": Object {
        "actions": Array [],
        "name": "lucas",
      },
    },
  },
}
`;
